connector/src/main/scala/com/datastax/spark/connector/ColumnSelector.scala

```diff
@@ -1,49 +1,70 @@
 package com.datastax.spark.connector
 
-import scala.language.implicitConversions
+import com.datastax.oss.driver.api.core.metadata.schema.TableMetadata
 
-import com.datastax.spark.connector.cql.TableDef
+import scala.language.implicitConversions
+import com.datastax.spark.connector.cql.{ColumnDef, StructDef, TableDef}
 
 sealed trait ColumnSelector {
   def aliases: Map[String, String]
   def selectFrom(table: TableDef): IndexedSeq[ColumnRef]
+  def selectFrom(table: TableMetadata): IndexedSeq[ColumnRef]
 }
 
 case object AllColumns extends ColumnSelector {
   override def aliases: Map[String, String] = Map.empty.withDefault(x => x)
   override def selectFrom(table: TableDef) =
     table.columns.map(_.ref)
+  override def selectFrom(table: TableMetadata) =
+    TableDef.columns(table).map(ColumnDef.toRef(_))
 }
 
 case object PartitionKeyColumns extends ColumnSelector {
   override def aliases: Map[String, String] = Map.empty.withDefault(x => x)
   override def selectFrom(table: TableDef) =
     table.partitionKey.map(_.ref).toIndexedSeq
+  override def selectFrom(table: TableMetadata) =
+    TableDef.partitionKey(table).map(ColumnDef.toRef(_)).toIndexedSeq
 }
 
 case object PrimaryKeyColumns extends ColumnSelector {
   override def aliases: Map[String, String] = Map.empty.withDefault(x => x)
   override def selectFrom(table: TableDef) =
     table.primaryKey.map(_.ref)
+  override def selectFrom(table: TableMetadata) =
+    TableDef.primaryKey(table).map(ColumnDef.toRef(_))
 }
 
 case class SomeColumns(columns: ColumnRef*) extends ColumnSelector {
 
+  private def columnsToCheck():Seq[ColumnRef] = columns flatMap {
+    case f: FunctionCallRef => f.requiredColumns //Replaces function calls by their required columns
+    case RowCountRef => Seq.empty //Filters RowCountRef from the column list
+    case other => Seq(other)
+  }
+
+  /** Compute the columns that are not present in the structure. */
+  private def missingColumns(table:StructDef, columnsToCheck: Seq[ColumnRef]): Seq[ColumnRef] =
+    for (c <- columnsToCheck if !table.columnByName.contains(c.columnName)) yield c
+
+  private def missingColumns(table:TableMetadata, columnsToCheck: Seq[ColumnRef]): Seq[ColumnRef] =
+    for (c <- columnsToCheck if !TableDef.containsColumn(c.columnName)(table)) yield c
```

Comment:
Formerly in StructDef.  Pulled out because it's only really used here so it seemed better to localize the functionality where it's needed.  If this becomes more broadly used it can always be moved back to StructDef